src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLoggerScope.cs

```diff
@@ -0,0 +1,75 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+#if NET451
+using System.Runtime.Remoting.Messaging;
+#else
+using System.Threading;
+#endif
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    public class EventFlowLoggerScope
+    {
+        internal EventFlowLoggerScope(object state)
+        {
+            State = state;
+        }
+
+        public Object State { get; private set; }
+        public EventFlowLoggerScope Parent { get; private set; }
+
+#if NET451
+        private static readonly string FieldKey = $"{typeof(EventFlowLoggerScope).FullName}";
```

Comment:
I followed what @lmolkova did for the Activity class.
Since we build provide Net451 and NetStandard1.1 packages, for the users who targets Net46, it will use the bits from Net451 package but not NetStandard1.1.
Considering the user base in full framework, I'd like to support the nested scope for 4.5.x as well.